mailserver/mailserver.go

```diff
@@ -53,9 +55,17 @@ var (
 	archivedErrorsCounter  = metrics.NewRegisteredCounter("mailserver/archiveErrors", nil)
 )
 
+type DB interface {
+	Close() error
+	Write(*leveldb.Batch, *opt.WriteOptions) error
```

Comment:
`Write` method is only used on the `cleaner.go` so you don't really need it on this interface.

Reply:
@adriacidre  in `cleaner_test.go` we take the `db` from the server and pass it to the cleaner. since `server.db` is the interface, that interface needs the `Write` method otherwise the cleaner can't call it.